### PR TITLE
Expose the keymap separately from the event keycode

### PR DESCRIPTION
**Pull request by slouken**
This adds functions to query the keymap:
* SDL_GetCurrentKeymap()
* SDL_GetKeymapKeycode()
* SDL_GetKeymapScancode()
* SDL_ReleaseKeymap()

and these are distinct from the function to query the event keycode associated with a scancode, which might be affected by SDL_HINT_KEYCODE_OPTIONS.

Also added an SDL_bool parameter to SDL_GetKeyName() and SDL_GetKeyFromName() to enable upper case handling of the name.

### CODE REVIEW

File: include/SDL3/SDL_keyboard.h [hunk 1]
```diff
@@ -185,80 +185,91 @@ extern SDL_DECLSPEC SDL_Keymod SDLCALL SDL_GetModState(void);
...
  * \since This function is available since SDL 3.0.0.
  *
- * \sa SDL_GetKeyName
- * \sa SDL_GetScancodeFromKey
+ * \sa SDL_GetKeymapKeycode
+ * \sa SDL_GetKeymapScancode
+ * \sa SDL_ReleaseKeymap
  */
-extern SDL_DECLSPEC SDL_Keycode SDLCALL SDL_GetDefaultKeyFromScancode(SDL_Scancode scancode, SDL_Keymod modstate);
+extern SDL_DECLSPEC SDL_Keymap * SDLCALL SDL_GetCurrentKeymap(void);
```
Comment (sezero) [inline include/SDL3/SDL_keyboard.h:210]: Should this return const ?

File: include/SDL3/SDL_keyboard.h [hunk 2]
```diff
@@ -185,80 +185,91 @@ extern SDL_DECLSPEC SDL_Keymod SDLCALL SDL_GetModState(void);
...
  * \since This function is available since SDL 3.0.0.
  *
- * \sa SDL_GetDefaultKeyFromScancode
- * \sa SDL_GetKeyName
- * \sa SDL_GetScancodeFromKey
+ * \sa SDL_GetCurrentKeymap
+ * \sa SDL_GetKeymapScancode
  */
-extern SDL_DECLSPEC SDL_Keycode SDLCALL SDL_GetKeyFromScancode(SDL_Scancode scancode, SDL_Keymod modstate);
+extern SDL_DECLSPEC SDL_Keycode SDLCALL SDL_GetKeymapKeycode(SDL_Keymap *keymap, SDL_Scancode scancode, SDL_Keymod modstate);
```
Comment (sezero) [inline include/SDL3/SDL_keyboard.h:228]: Should this accept const SDL_Keymap* ? (see above)

File: include/SDL3/SDL_keyboard.h [hunk 3]
```diff
@@ -185,80 +185,91 @@ extern SDL_DECLSPEC SDL_Keymod SDLCALL SDL_GetModState(void);
...
  *
  * \since This function is available since SDL 3.0.0.
  *
- * \sa SDL_GetScancodeFromKey
- * \sa SDL_GetScancodeName
+ * \sa SDL_GetCurrentKeymap
+ * \sa SDL_GetKeymapKeycode
  */
-extern SDL_DECLSPEC SDL_Scancode SDLCALL SDL_GetDefaultScancodeFromKey(SDL_Keycode key, SDL_Keymod *modstate);
+extern SDL_DECLSPEC SDL_Scancode SDLCALL SDL_GetKeymapScancode(SDL_Keymap *keymap, SDL_Keycode keycode, SDL_Keymod *modstate);
```
Comment (sezero) [inline include/SDL3/SDL_keyboard.h:247]: ditto

File: include/SDL3/SDL_keyboard.h (hunk 1 above)
Comment (sezero) [reply include/SDL3/SDL_keyboard.h:210]: Hm, but will SDL_ReleaseKeymap() be problem with const?

**Comment by sezero** [comment]
sdl2-compat need adjusting after this.

**Comment by slouken** [comment]
I'm rolling this back. I have an sdl2-compat update coming.

**Pull request merged**